### PR TITLE
Adding glouonts.torch.piecewise_linear

### PR DESCRIPTION
**Pull request by KelvinKan**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by kashif** [comment]
nice! @KelvinKan can you kindly add some tests? Feel free to use: https://github.com/zalandoresearch/pytorch-ts/blob/master/test/distributions/test_piecewise_linear.py

File: src/gluonts/torch/modules/piecewise_linear.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    DistributionOutput,
+)
+
+from torch.distributions import (
+    AffineTransform,
+    TransformedDistribution,
+)
+
+
+class PiecewiseLinear(Distribution):
```
Comment (lostella) [inline src/gluonts/torch/modules/piecewise_linear.py:31]: Analogously to the structure of `torch`, that keeps distributions classes within `torch.distributions`, I would put this whole module into `gluonts.torch.distributions`

File: src/gluonts/torch/modules/piecewise_linear.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    def get_u_tilde(self, z: torch.Tensor) -> torch.Tensor:
+        """
+        compute the quantile levels u_tilde s.t. quantile(u_tilde)=z
+
+        Input
+        z: observations, shape = gamma.shape = (*batch_size)
+
+        Output
+        u_tilde: of type torch.Tensor
+        """
```
Comment (lostella) [inline src/gluonts/torch/modules/piecewise_linear.py:121]: The docs build logs say
```
Warning, treated as error:
/home/runner/work/gluon-ts/gluon-ts/src/gluonts/torch/modules/piecewise_linear.py:docstring of gluonts.torch.modules.piecewise_linear.PiecewiseLinear.get_u_tilde:3:Inline emphasis start-string without end-string.
```

So it looks like they don't like this docstring. You can try to use the same formatting as other docstrings in the package:

```
"""
<Short description>

<Optional long description>

Parameters
----------
<arg-name-1>
    <description>
<arg-name-2>
    <description>
...
Returns
-------
<return type>
    <description>
```

and wrapping any python code in double-backticks (`)

**Comment by lostella** [comment]
@KelvinKan the style checks are failing: to fix these, make sure you have the right `black` version installed with the following

```
pip install requirements/requirements-setup.txt
```

then you can fix the offending files with 

```
black src test
```

and committing the changes that `black` applies. I advise you to run the following command to set up some git pre-commit hooks that run the checks locally, every time you commit, so that you know in advance whether some checks are failing:

```
./dev_setup.sh
```

**Comment by lostella** [comment]
Closing in favor of #1663

**Pull request closed without merging**